src/main/scala/Config.scala

```diff
@@ -185,15 +186,25 @@ class Config(args: Seq[String]) extends SilFrontendConfig(args, "Silicon") {
     valueName = "level"
   )
 
-  val timeout: ScallopOption[Int] = opt[Int]("timeout",
+  private val rawTimeout: ScallopOption[Int] = opt[Int]("timeout",
     descr = ( "Time out after approx. n seconds. The timeout is for the whole verification, "
-            + "not per method or proof obligation (default: 0, i.e. no timeout)."),
-    default = Some(0),
+            + s"not per method or proof obligation (default: 0 i.e. no timeout for all provers except ${Cvc5ProverStdIO.name} with default 180)."),
+    default = None,
     noshort = true
   )
 
+  lazy val timeout: Int = rawTimeout.toOption match {
+    case Some(t) => t
+    // Not set, so compute default value based on prover selected.
+    case _ => prover.toOption match {
+      case Some(Cvc5ProverStdIO.name) => 180
```

Comment:
Why? It seems strange to have a different default based on the prover. The description of this in line 191 is also not great.

Reply:
I changed the description in commit 2924e00fec1cc6b264f66840a4ac775ca434e3f3 and added a comment to the file.

The default timeout for cvc5 is set to 180 since it does not support per-checks timeouts set after instantiation of the solver so any checks or saturation checks can cause the verifier to hang indefinitely.
But maybe we just let the default be 0 for all provers and we let cvc5 users deal with this themselves?

I appreciate any input on this :)

Reply:
I don't have a strong opinion here, but it seems cleaner to not "misuse" the global timeout to compensate for cvc5's lack of per-check timeouts. Please revert to a default of 0 for all provers.

Reply:
Fixed/reverted back to default of 0 for all provers in 37e3adb6f12fbd620710edc401ecd78ef87b5bb7, 6ab985857e5f8dafa1e3645d6b110d8075e7bb00, and 4215b09691a7a66163388c4e562b5b00f9941c7e.